### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# How to contribute to proj-datumgrid
+
+Consult [PROJ CONTRIBUTING.md](https://github.com/OSGeo/proj.4/blob/master/CONTRIBUTING.md)
+for general principles.
+
+This document focuses on how to specifically contribute a new grid to proj-datumgrid.
+
+Preliminary:
+* The grid must be in one of the formats understood by PROJ, typically .gsb for horizontal shift
```
Comment (kbevers) [inline CONTRIBUTING.md]: The format is called NTv2, isn't it?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+grids and .gtx for vertical shift grids
+* The grid must be freely licensed as mentionned in [README.DATUMGRID](README.DATUMGRID).
+* We consider that you are familiar with the basic [GitHub workflow to submit a pull request](https://help.github.com/en/articles/creating-a-pull-request)
+
+Determine the geographic scope of the grid. We have one sub directory for
+continental packages
+- europe
+- north-america
+- oceania
```
Comment (kbevers) [inline CONTRIBUTING.md:18]: There's also "world"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+In the case of overseas territory, we have currently made the choice to attach grids
+to the continent of the mainland (hence all grids for French territories are in europe/,
+for USA in north-america/ etc.)
+
+Let's consider the case of adding a grid for a new European country.
+The steps are:
+1. Add it in the europe/ directory
```
Comment (kbevers) [inline CONTRIBUTING.md]: Formatting: `europe/`. Add some `'s around directory and file names. Improves readability.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+already have an entry for the grid, sometimes with a slightly different name.
+The relevant file to look into is [grid_transformation.sql](https://github.com/OSGeo/proj.4/blob/master/data/sql/grid_transformation.sql)
+
+You may find an entry like the following one:
+```
+INSERT INTO "grid_transformation" VALUES('EPSG','15958','RGF93 to NTF (2)',NULL,NULL,'EPSG','9615','NTv2','EPSG','4171','EPSG','4275','EPSG','3694',1.0,'EPSG','8656','Latitude and longitude difference file','rgf93_ntf.gsb',NULL,NULL,NULL,NULL,NULL,NULL,'ESRI-Fra 1m emulation',0);
+```
+This is a transformation from EPSG:4171 (RGF93) to EPSG:4275 (NTF) using the rgf93_ntf.gsb grid.
+
+In the case the EPSG dataset does not include an entry, a custom entry may be added in the [grid_transformation_custom.sql](https://github.com/OSGeo/proj.4/blob/master/data/sql/grid_transformation_custom.sql) file.
```
Comment (kbevers) [inline CONTRIBUTING.md]: "in *this* case..."?

**Review by kbevers — COMMENTED** [review]
Looks good to me, only a few minor comments. I even learned a thing or two while reading it :-)

File: CONTRIBUTING.md (hunk 4 above)
Comment (rouault) [reply CONTRIBUTING.md]: Frenchism...I meant "if".

**Pull request merged**